src/rpc/RPCHelpers.cpp

```diff
@@ -569,7 +569,7 @@ ledgerInfoFromRequest(Context const& ctx)
 
         auto lgrInfo = ctx.backend->fetchLedgerByHash(ledgerHash, ctx.yield);
 
-        if (!lgrInfo)
+        if (!lgrInfo || lgrInfo->seq > ctx.range.maxSequence)
```

Comment:
Do we not perform this check earlier? If we don't, I think we need to. What if users request data from a specific ledger, but we don't actually fetch the ledger header?

Reply:
I think we always fetch the ledger header to get the ledger sequence

Reply:
`ledgerInfoFromRequest` is called in every RPC.